src/evict/evict_page.c

```diff
@@ -557,15 +564,27 @@ __evict_child_check(WT_SESSION_IMPL *session, WT_REF *parent)
      * opposite direction from our check.
      */
     WT_INTL_FOREACH_BEGIN (session, parent->page, child) {
+        /* It isn't safe to evict if there is a child on the pre-fetch queue. */
```

Comment:
Is it reasonable to reuse one of the other "look at all the children" loops in this function?

Reply:
I'm not too sure what you mean here.

Reply:
I think it's already layered into the forward pass - but the diff is a bit hard to interpret.

Reply:
This function has a whole series of `WT_INTL_FOREACH_BEGIN` (and reverse versions) to iterate over the child nodes. I'm not sure how much sense it makes to add another one as opposed to reusing one of the existing loops.

Reply:
Oh right - didn't see Alex's comment while I was typing it out. If I misread the diff and it's already in one of the passes, feel free to close out this thread.